### PR TITLE
UI

### PR DESCRIPTION
**Pull request by alfredodeza**
Initial take on a UI for shaman

### CODE REVIEW

File: public/js/morrisjs/morris-data.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            calamari: null,
+            diamond: 1
+        }, {
+            period: '2016-09-7',
+            ceph: 7,
+            calamari: 2,
+            diamond: 2
+        }, {
+            period: '2016-09-8',
+            ceph: 4,
```
Comment (andrewschoen) [inline public/js/morrisjs/morris-data.js]: I assume these projects will be data driven at some point?

File: shaman/controllers/builds/projects.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            self.project = models.get_or_create(Project, name=self.project_name)
+        chacra_url = request.json["chacra_url"]
+        repo = Build.query.filter_by(chacra_url=chacra_url).first()
+        if not repo:
+            data = dict(
+                project=self.project,
+                ref=request.json["ref"],
+                sha1=request.json["sha1"],
+                flavor=request.json["flavor"],
+                distro=request.json["distro"],
```
Comment (andrewschoen) [inline shaman/controllers/builds/projects.py]: ``distro``, ``distro_version`` and ``arch`` are no longer part of the ``Build`` model

File: shaman/controllers/builds/__init__.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            abort(404, 'project not found')
+        request.context['flavor'] = flavor_name
+
+    @expose('jinja:builds.html')
+    def index(self):
+        builds = Build.filter_by(
+            project=self.project,
+            ref=request.context['ref'],
+            sha1=request.context['sha1'],
+            distro=request.context['distro'],
```
Comment (andrewschoen) [inline shaman/controllers/builds/__init__.py]: ``distro``, ``distro_version`` and ``arch`` are no longer part of the ``Build`` model

File: shaman/controllers/builds/__init__.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            # TODO: nice project not found error template
+            abort(404, 'project not found')
+
+    @expose('jinja:build.html')
+    def index(self):
+        build = Build.filter_by(
+            project=self.project,
+            ref=request.context['ref'],
+            sha1=request.context['sha1'],
+            distro=request.context['distro'],
```
Comment (andrewschoen) [inline shaman/controllers/builds/__init__.py]: ``distro``, ``distro_version`` and ``arch`` are no longer part of the ``Build`` model

File: shaman/controllers/builds/projects.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            set([r.ref for r in self.project.repos])
+        )
+
+    #TODO: we need schema validation on this method
+    @secure(basic_auth)
+    @index.when(method='POST', template='json')
+    def index_post(self):
+        if not self.project:
+            self.project = models.get_or_create(Project, name=self.project_name)
+        chacra_url = request.json["chacra_url"]
```
Comment (andrewschoen) [inline shaman/controllers/builds/projects.py]: I think you want ``build_url`` here, not ``chacra_url``

File: shaman/controllers/builds/projects.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        repo = Build.query.filter_by(chacra_url=chacra_url).first()
```
Comment (andrewschoen) [inline shaman/controllers/builds/projects.py]: Same, ``build_url`` not ``chacra_url``

File: shaman/models/builds.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+from sqlalchemy.orm.exc import DetachedInstanceError
+from shaman.models import Base, update_timestamp
+from shaman.models.types import JSONType
+
+
+class Build(Base):
+
+    __tablename__ = 'builds'
+    id = Column(Integer, primary_key=True)
+    url = Column(String(256))
```
Comment (andrewschoen) [inline shaman/models/builds.py]: What's the difference between ``url`` and ``build_url``?

File: public/js/morrisjs/morris-data.js (hunk 1 above)
Comment (alfredodeza) [inline public/js/morrisjs/morris-data.js]: oi. Have to remove these

File: shaman/models/builds.py (hunk 7 above)
Comment (alfredodeza) [inline shaman/models/builds.py]: hrmn, this `build_url` was initially `log_url`. 

`url` was meant a `url` pointing back to the job url (not the log)

File: shaman/controllers/builds/projects.py (hunk 6 above)
Comment (alfredodeza) [inline shaman/controllers/builds/projects.py]: fixed

File: shaman/controllers/builds/projects.py (hunk 2 above)
Comment (alfredodeza) [inline shaman/controllers/builds/projects.py]: fixed

File: shaman/controllers/builds/__init__.py (hunk 4 above)
Comment (alfredodeza) [inline shaman/controllers/builds/__init__.py]: fixed

File: shaman/controllers/builds/__init__.py (hunk 3 above)
Comment (alfredodeza) [inline shaman/controllers/builds/__init__.py]: fixed

File: shaman/models/builds.py (hunk 7 above)
Comment (alfredodeza) [inline shaman/models/builds.py]: made this url and log_url

**Comment by alfredodeza** [comment]
test this please

**Pull request merged**